internal/ingress/controller/parser/ingressrules.go

```diff
@@ -80,22 +82,19 @@ func newSecretNameToSNIs() SecretNameToSNIs {
 	return SecretNameToSNIs(map[string][]string{})
 }
 
-func (m SecretNameToSNIs) addFromIngressV1beta1TLS(tlsSections []networkingv1beta1.IngressTLS, namespace string) {
-	for _, tls := range tlsSections {
-		if len(tls.Hosts) == 0 {
-			continue
-		}
-		if tls.SecretName == "" {
-			continue
-		}
-		hosts := tls.Hosts
-		secretName := namespace + "/" + tls.SecretName
-		hosts = m.filterHosts(hosts)
-		if m[secretName] != nil {
-			hosts = append(hosts, m[secretName]...)
-		}
-		m[secretName] = hosts
+func (m SecretNameToSNIs) addFromIngressV1beta1TLS(tlsSections []networkingv1beta1.IngressTLS, namespace string) error {
+	// Assume that v1beta1 and v1 tlsSections have identical semantics and field-wise content.
+	var bytes []byte
+	var err error
+	if bytes, err = json.Marshal(tlsSections); err != nil {
+		return err
+	}
```

Comment:
We should not be resorting to JSON marshal/unmarshal here.
Can we not iterate over TLS structures and copy them over natively?

Reply:
Done